### PR TITLE
Add multi-step-flow sample

### PR DESCRIPTION
**Pull request by atanasiuk-hubspot**
Copy of private apps [sample](https://github.com/HubSpot/ui-extensions-examples/tree/main/multi-step-flow) migrated to public apps

сс @aanchalHS

### CODE REVIEW

**Review by skaplan-dev — APPROVED** [review]

File: multi-step-flow/src/app/app.functions/package.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "hubspot-example-function",
+  "version": "0.1.0",
+  "author": "HubSpot",
+  "license": "MIT",
+  "dependencies": {
+    "@hubspot/api-client": "^7.0.1",
+    "axios": "^0.27.2"
+  }
+}
```
Comment (dshekhtman) [inline multi-step-flow/src/app/app.functions/package.json]: Need a new line at the end here.

File: multi-step-flow/src/app/app.functions/restaurants.js [hunk 2]
```diff
@@ -0,0 +1,189 @@
+// Credit to Dan Dascalescu and Community:
+// https://stackoverflow.com/questions/951021/what-is-the-javascript-version-of-sleep
+const sleep = (ms) => new Promise((r) => setTimeout(r, ms));
```
Comment (dshekhtman) [inline multi-step-flow/src/app/app.functions/restaurants.js]: Can we change `r` to `resolve` so this is a little easier to understand?

File: multi-step-flow/src/app/extensions/components/OrderMealCard.tsx [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+  const uniqueRestaurants = new Set(cart.map((item) => item.restaurantId));
+  const totalDeliveryCost = [...uniqueRestaurants].reduce((total, id) => {
+    return total + (getRestaurant(id)?.deliveryCost ?? 0);
+  }, 0);
+
+  const selectedRestaurant = getRestaurant(selected);
+  const subtotal = cart.reduce((total, item) => total + item.price, 0);
+
+  return (
+    <Flex direction={'column'} gap={'md'}>
```
Comment (dshekhtman) [inline multi-step-flow/src/app/extensions/components/OrderMealCard.tsx]: Super nit, but these can be `direction="column" gap="md"`. There's a couple other instances of this around.

**Review by dshekhtman — APPROVED** [review]
LGTM otherwise!

**Pull request merged**